### DISCUSSION TITLE
_TZE200_mudxchsu and schedule

### DISCUSSION THREAD

**Discussion by git-tail-d** [Koenkk/zigbee2mqtt#19122] [Devices]
Hello,

I bought the Tuya heating controller _TZE200_mudxchsu a few months ago and tested it successfully.

Now when I want to change the automatic switching times and select the day of the week, I get the following error: Publish 'set' 'schedule' to '0x0c4314fffef335a8' failed: 'TypeError: Cannot read properties of undefined (reading 'toString')'.

To rule out possible errors in the configuration, I set up a Raspberry3 with openHABian, Mosquitto and Z2M again. But I get the same error there too.

Current versions:
mosquitto version 2.0.11
Z2M: 1.33.0 commit: 4d53ac4c
openHAB 4.0.3
Coordinator-Typ     ConBee2/RaspBee2
Coordinator-Version     0x26720700

Best Regards

**Comment by git-tail-d** [comment]
It is same like https://github.com/Koenkk/zigbee2mqtt/issues/15306

**Comment by git-tail-d** [comment]
Hello,

here is the errormessage:
`error 2023-10-15 10:27:03Publish 'set' 'schedule' to 'Heiz_EDV' failed: 'TypeError: Cannot read properties of undefined (reading 'toString')'
debug 2023-10-15 10:27:03TypeError: Cannot read properties of undefined (reading 'toString') at Object.assertString (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/src/lib/utils.ts:501:111) at Object.to (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/src/lib/tuya.ts:634:19) at Object.convertSet (/opt/zigbee2mqtt/node_modules/zigbee-herdsman-converters/src/lib/tuya.ts:883:57) at Publish.onMQTTMessage (/opt/zigbee2mqtt/lib/extension/publish.ts:248:52) at EventEmitter.emit (node:events:526:35) at EventBus.emitMQTTMessage (/opt/zigbee2mqtt/lib/eventBus.ts:109:22) at MQTT.onMessage (/opt/zigbee2mqtt/lib/mqtt.ts:141:27) at WebSocket.<anonymous> (/opt/zigbee2mqtt/lib/extension/frontend.ts:127:27) at WebSocket.emit (node:events:514:28) at Receiver.receiverOnMessage (/opt/zigbee2mqtt/node_modules/ws/lib/websocket.js:1192:20)`

**Comment by Koenkk** [comment]
See https://github.com/Koenkk/zigbee2mqtt/issues/19240#issuecomment-1763467981
